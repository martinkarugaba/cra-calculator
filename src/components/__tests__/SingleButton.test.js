import React from "react";
import { render } from "@testing-library/react";
import SingleButton from "../SingleButton";
import AppContext from "../../context/context";

describe("SingleButton component", () => {
  test("renders correctly", () => {
    const props = {
      buttonName: "Test Button",
      basis: "basis-class",
      bg: "bg-class",
    };

    const { container } = render(
      <AppContext>
        <SingleButton {...props} />
      </AppContext>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
