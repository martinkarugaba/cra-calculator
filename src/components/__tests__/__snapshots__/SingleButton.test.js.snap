// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleButton component renders correctly 1`] = `
<button
  class="basis-class flex justify-center items-center bg-class hover:bg-slate-400 text-black h-[60px] border-r border-t text-xs cursor-pointer"
  type="button"
>
  Test Button
</button>
`;
